btree/btree.c

```diff
@@ -3654,25 +3658,29 @@ int64_t btree_destroy_tick(struct m0_sm_op *smop)
 		 * updated.
 		 */
 		m0_free(bop->bo_arbor);
-		m0_free(bop->bo_i);
 		bop->bo_arbor = NULL;
-		bop->bo_i = NULL;
 
-		return P_DONE;
+		/** The check for rc has added to gracefully handle the case
+		 *  where the t_ref is 0. If we had put an assert there, then
+		 * we will face memory leakage in terms of bop->bo_arbor.
+		 */
+		if(rc == 0)
+			return P_DONE;
+		return rc;
```

Comment:
`rc` should not be returned. It should be set in `sm_rc`

Reply:
I think we should return the error as in m0_sm_op_tick()
![image](https://user-images.githubusercontent.com/62925646/123374695-27dc4800-d5a5-11eb-8ac0-d18677e5bd71.png)
So the error returned will be set op->o_sm.sm_rc

Reply:
Maybe I will check again.

Reply:
I confirmed this.
We cannot use sm_rc. I say this because the state we will be returning after setting sm_rc would be P_DONE or P_CLEANUP. As neither of them are negative, we wont go inside the result < 0 block. If it doesnt go inside that block, it means that fail wont be true. Therefore, m0_sm_state_set() will reset the sm_rc value to 0.

Reply:
Ok. On further checking the code, `rc = 0` check is added just for txref = 0, which is an invalid condition. You should have an assert to check this invalid scenario.